lib/deployment.js

```diff
@@ -0,0 +1,156 @@
+        let event_action = this.actionContext.github.event.action
+
+        // PR Values (if applicable)
+        let pull_request = this.actionContext.github.event.pull_request
+        let pr_number = pull_request !== undefined ? pull_request.number : ""
+        let pr_suffix = pr_number !== "" ? "-pr-" + pr_number : ""
+        let pr_label = ""
+        if (this.actionContext.github.event.label) {
+            pr_label = this.actionContext.github.event.label.name // Only present on label / unlabeled events
+        }
+        let pr_closed = event === "pull_request" && event_action === "closed"
+
+        // Git Information
+        let git_ref = this.actionContext.github.head_ref !== "" ? this.actionContext.github.head_ref : this.actionContext.github.ref_name
+        let git_sha = event === "pull_request" ? pull_request.head.sha : this.actionContext.github.sha
+
+        // Preview Environment Values
+        let preview_labeled = pull_request !== undefined ? pull_request.labels.some(function (element) {
+            return element.name === "preview";
+        }) : false
+        let preview_unlabeled = event_action === "unlabeled" && pr_label === "preview"
+
+        // CI/CD Values
+        let image_build_enabled = !preview_unlabeled && !pr_closed
+        let deployment_enabled  = this.actionContext.inputs.deploy === "true" && image_build_enabled
+
+        // Deploy Action
+        let deployment_action = ""
+        if (pr_closed || preview_unlabeled) {
+            deployment_action = "delete"
+        } else if (preview_labeled || event === "push" || event === "release") {
+            deployment_action = "create"
+        }
+
+        return {
+            event: event,
+            event_action: event_action,
+            git_ref: git_ref,
+            git_sha: git_sha,
+            repo_name: this.actionContext.context.repo.repo,
+            pr_number: pr_number,
+            pr_suffix: pr_suffix,
+            pr_label: pr_label,
+            pr_closed: pr_closed,
+            preview_labeled: preview_labeled,
+            preview_unlabeled: preview_unlabeled,
+            image_build_enabled: image_build_enabled,
+            deployment_enabled: deployment_enabled,
+            deployment_action: deployment_action,
+        }
+    }
+
+    // Note that deployment_action in the context of CD is not 1-1 to the same variable in the CI context, because the GHA
+    // workflow is triggered by both deployment and deployment_status.
+    getInfoCD() {
+        let deployment_action = ""
+        if (this.actionContext.github.event_name !== 'deployment_status' && this.actionContext.github.event.deployment.payload.deploy_action === 'create') {
+            deployment_action = "create"
+        } else if (this.actionContext.github.event_name === 'deployment_status' && this.actionContext.github.event.deployment_status.description === 'delete') {
+            deployment_action = "delete"
```

Comment:
> Why use a string here instead of a boolean ?

I thought the same thing, I just didn't think about changing it as it was being ported. I'll take a closer look at converting the strings into booleans.